homeassistant/components/bluetooth_tracker/device_tracker.py

```diff
@@ -104,17 +119,20 @@ async def get_tracking_devices(hass: HomeAssistant) -> tuple[set[str], set[str]]
 def lookup_name(mac: str) -> str | None:
     """Lookup a Bluetooth device name."""
     _LOGGER.debug("Scanning %s", mac)
-    return bluetooth.lookup_name(mac, timeout=5)
+    return bluetooth.lookup_name(mac, timeout=5)  # type: ignore[no-any-return]
 
 
 async def async_setup_scanner(
-    hass: HomeAssistant, config: dict, async_see, discovery_info=None
-):
+    hass: HomeAssistant,
+    config: ConfigType,
+    async_see: Callable[..., None],
+    discovery_info: dict[str, Any] | None = None,
+) -> bool:
     """Set up the Bluetooth Scanner."""
     device_id: int = config[CONF_DEVICE_ID]
-    interval = config.get(CONF_SCAN_INTERVAL, SCAN_INTERVAL)
-    request_rssi = config.get(CONF_REQUEST_RSSI, False)
-    update_bluetooth_lock = asyncio.Lock()
+    interval: timedelta = config.get(CONF_SCAN_INTERVAL, SCAN_INTERVAL)
+    request_rssi: bool = config.get(CONF_REQUEST_RSSI, False)
+    update_bluetooth_lock: Lock = asyncio.Lock()
```

Comment:
This is excessive, automatic inference will handle it. 
```suggestion
    update_bluetooth_lock = asyncio.Lock()
```